shared/utils/index.ts

```diff
@@ -112,3 +112,89 @@ export function checkHrefFormat(href: string): string {
 
   return href
 }
+
+export class WalletValidationError extends Error {
+  constructor(message: string, options?: { cause?: unknown }) {
+    super(message, { cause: options?.cause })
+    this.name = 'WalletValidationError'
+  }
+}
+
+export async function fetchWalletDetails(url: string): Promise<WalletAddress> {
+  const res = await fetch(url, {
+    headers: { Accept: 'application/json' }
+  })
+  if (!res.ok) {
+    throw new WalletValidationError('is not a valid wallet address')
+  }
+  return await res.json()
+}
+
+export async function confirmWalletAddress(
+  walletAddress: string
+): Promise<void> {
+  const wallet = await fetchWalletDetails(walletAddress)
+
+  if (!isWalletAddress(wallet)) {
+    throw new WalletValidationError('does not have a valid wallet response')
+  }
+}
+
+export const validateWalletAddressOrPointer = (input: string): string => {
+  if (!input || typeof input !== 'string') {
+    throw new WalletValidationError('must be a string')
+  }
```

Comment:
Users might not understand what this error means. Maybe saying this field is required is better (or we can even mark the input as `required` so we get native form validation).

<img width="838" height="551" alt="image" src="https://github.com/user-attachments/assets/d9384e78-e6c3-4c0d-9317-512b0f8a6928" />